src/rbyte/utils/functional.py

```diff
@@ -0,0 +1,41 @@
+from collections.abc import Sequence
+
+import more_itertools as mit
+import torch
+import torch.nn.functional as F  # noqa: N812
+from jaxtyping import Float
+from torch import Tensor
+
+
+def pad_dim(
+    input: Float[Tensor, "..."],  # noqa: A002
+    *,
+    pad: tuple[int, int],
+    dim: int,
+    mode: str = "constant",
+    value: float | None = None,
+) -> Float[Tensor, "..."]:
+    _pad = [(0, 0) for _ in input.shape]
+    _pad[dim] = pad
+    _pad = list(mit.flatten(reversed(_pad)))
+
+    return F.pad(input, _pad, mode=mode, value=value)
+
+
+def pad_sequence(
```

Comment:
https://pytorch.org/docs/stable/generated/torch.nn.utils.rnn.pad_sequence.html

Reply:
was my first thought/attempt as well but need a slightly more flexible api: specifying which dim to pad